File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/collection/model/CollectionTranslationData.java
package dev.reuss.tmdb.domain.collection.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Translated collection data.
 *
 * @param title    translated title
 * @param overview translated overview
 * @param homepage translated homepage
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record CollectionTranslationData(
        String title,
        String overview,
        String homepage
) {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/collection/model/CollectionTranslationData.kt
package dev.reuss.tmdb.domain.collection.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Translated collection data.
 *
 * @property title translated title
 * @property overview Localized overview when available.
 * @property homepage translated homepage
 */
@JvmRecord
data class CollectionTranslationData(
    @all:JsonProperty("title")
    val title: String?,
    @all:JsonProperty("overview")
    val overview: String?,
    @all:JsonProperty("homepage")
    val homepage: String?,
) : TmdbModel
